integration/testrunner/http.go

```diff
@@ -64,18 +75,63 @@ func testHTTPEndpoint(ctx context.Context, tb testing.TB, endpointURL, idToken,
 	}); err != nil {
 		tb.Fatal(err)
 	}
-
 	bqClient := makeBigQueryClient(ctx, tb, projectID)
-
+	time.Sleep(10 * time.Second)
 	bqQuery := makeQueryForHTTP(*bqClient, id, projectID, datasetQuery)
-	queryIfAuditLogExistsWithRetries(ctx, tb, bqQuery, cfg, "httpEndpointTest")
+	tb.Log(bqQuery.Q)
+	results := queryAuditLogsWithRetries(ctx, tb, bqQuery, cfg, "httpEndpointTest")
+	wantNum := 1
+	if len(results) != wantNum {
+		tb.Errorf("log number doesn't match (-want +got):\n - %d\n + %d\n", wantNum, len(results))
+	} else {
+		jsonPayloadInfo := parseJsonpayload(tb, results[0])
+		diffResults(tb, results[0], jsonPayloadInfo, isHTTP("HTTP"), 0)
+	}
 }
 
 func makeQueryForHTTP(client bigquery.Client, id, projectID, datasetQuery string) *bigquery.Query {
-	// Cast to int64 because the result checker expects a number.
-	queryString := fmt.Sprintf("SELECT CAST(EXISTS (SELECT * FROM `%s.%s` WHERE labels.trace_id=?", projectID, datasetQuery)
-	queryString += ` AND jsonPayload.service_name IS NOT NULL`
-	queryString += ` AND jsonPayload.authentication_info.principal_email IS NOT NULL`
-	queryString += ") AS INT64)"
+	queryString := fmt.Sprintf(` WITH temptable AS (
+		SELECT *
+		FROM %s.%s
+		WHERE labels.trace_id=?
+ 	)
+ 	SELECT TO_JSON(t) as result FROM temptable as t
+	`, projectID, datasetQuery)
 	return makeQuery(client, id, queryString)
 }
+
+func diffResults(tb testing.TB, logEntry *logpb.LogEntry, jsonPayloadInfo *audit.AuditLog, isHTTPService bool, index int) {
```

Comment:
instead of passing jsonPayloadInfo as a parameter, can we just pass logEntry and call `parsingJsonPayload` in this function and get the jsonPayloadInfo?

Reply:
Done